### DISCUSSION TITLE
RuntimeError: asyncio.run() cannot be called from a running event loop

### DISCUSSION THREAD

**Discussion by andysingal** [safevideo/autollm#138] [Q&A] [answered]
Notebook: https://colab.research.google.com/github/safevideo/autollm/blob/main/examples/quickstart.ipynb

<img width="600" alt="Screenshot 2023-11-07 at 2 37 20 PM" src="https://github.com/safevideo/autollm/assets/20493493/687b0297-bafb-4afa-a5f9-7b21bb560f78">

**Comment by talhasarit** [comment, marked as answer]
Hello @andysingal 

Thank you for using `autollm` and for reporting this issue.

The error is due to the attempt to start an async `uvicorn` server in Google Colab, which already runs in an async event loop. To resolve this, use `nest_asyncio` to patch the event loop. Here’s the code snippet you can use:

```python
!pip install nest_asyncio
import nest_asyncio
from autoLm import AutoFastAPI

nest_asyncio.apply()
app = AutoFastAPI.from_query_engine(query_engine)

# Now you can run the app without the RuntimeError
uvicorn.run(app, host="0.0.0.0", port=8000)
```

We hope this helps, and we’re here for any further questions or feedback.

Best,
Talha
